repo_policy_compliance/__init__.py

```diff
@@ -93,57 +95,116 @@ def _check_signed_commits_required(branch: Branch) -> Report:
     return Report(result=Result.PASS, reason=None)
 
 
-def all_(
-    repository_name: str,
-    source_repository_name: str,
-    target_branch_name: str,
-    source_branch_name: str,
-    commit_sha: str,
-) -> Report:
-    """Run all the checks.
+class UsedPolicy(Enum):
+    """Sentinel to indicate which policy to use.
 
-    Args:
+    Attrs:
+        ALL: Use all policies.
+    """
+
+    ALL = 1
+
+
+class Input(NamedTuple):
+    """Input arguments for checks.
+
+    Attrs:
         repository_name: The name of the repository to run the check on.
         source_repository_name: The name of the repository that has the source branch.
         target_branch_name: The name of the branch that is targeted by the PR.
         source_branch_name: The name of the branch that contains the commits to be merged.
         commit_sha: The SHA of the commit that the workflow run is on.
+    """
+
+    repository_name: str
+    source_repository_name: str
+    target_branch_name: str
+    source_branch_name: str
+    commit_sha: str
+
+
+def _policy_enabled(name: str, policy_document: MappingProxyType) -> bool:
+    """Check whether a given policy is enabled.
+
+    Args:
+        name: The property name of the policy.
+        policy_document: Describes the policies that should be run.
+
+    Returns:
+        Whether the policy is enabled in the document.
+    """
+    return name in policy_document and policy_document[name]["enabled"]
+
+
+def all_(input_: Input, policy_document: dict | UsedPolicy = UsedPolicy.ALL) -> Report:
+    """Run all the checks.
+
+    Args:
+        input_: Data required for executing checks.
+        policy_document: Describes the policies that should be run.
 
     Returns:
         Whether the run is authorized based on all the checks.
     """
+    if policy_document == UsedPolicy.ALL:
+        used_policy_document: MappingProxyType = policy.ALL
+    else:
+        # Guaranteed to be a dict due to initial if
```

Comment:
It seems `policy_document` is of type `dict | UsedPolicy`.  Can this typing be improved? E.g., more specific type of dict.

Reply:
the intent for it is to be an arbitrary dict and it gets checked against the specific schema to make sure that it is what is expected, does that make sense?

Reply:
So line 154 checks that the dictionary is a valid policy document